Openscapes Workshop Blog post (#274)
2i2c-org/2i2c-brochure: content/blog/2024/openscapes-sbg-workshop/index.md

---

> title: Openscapes host a Surface Biology and Geology workshop

Reviewer: ```suggestion
title: Openscapes Host a Surface Biology and Geology Workshop with Shared Password
```

date: "2024-05-29"
banner:
  image: "featured.png"
authors: ["Yuvi Panda", "Jenny Wong"]
tags: [geoscience, education]
categories: [impact, updates, partnerships]
featured: false
draft: false
---

![Cover slide from the SBG Workshop](featured.png "Surface Biology and Geology: VITALS Workshop")

> [Openscapes](https://nasa-openscapes.github.io/) is a value-based initiative that advocates for kinder, better science produced with open source community and program building co-developed by Julia Lowndes and Erin Robinson. 

Reviewer 2: I wonder if @jules32 has a sentence to add about the NASA+Openscapes parnership?

Reviewer: Thanks Andy, I've added suggested text.

Author: Resolved in https://github.com/2i2c-org/2i2c-org.github.io/pull/274/commits/2ac81f2aee48b29c9a8cd431663521739f9ee6a5

> [Openscapes](https://nasa-openscapes.github.io/) is a value-based initiative that advocates for kinder, better science produced with open source community and program building co-developed by Julia Lowndes and Erin Robinson. 

Reviewer: ```suggestion
[Openscapes](https://openscapes.org) is a value-based initiative that supports kinder, better science produced with open source community. [NASA Openscapes](https://nasa-openscapes.github.io) is in its 4th year as a project supporting NASA Earth science in the Cloud, co-developed by Julia Lowndes (Openscapes) and Erin Robinson (Metadata Game Changers). 
```

Author: ```suggestion
[Openscapes](https://openscapes.org) is a value-based initiative that supports kinder, better science based on open source community. [NASA Openscapes](https://nasa-openscapes.github.io) is in its fourth year as a project supporting NASA Earth science in the Cloud, co-developed by Julia Lowndes (Openscapes) and Erin Robinson (Metadata Game Changers).  
```

The initiative recently supported the [Surface Biology and Geology: VITALS Workshop](https://nasa.github.io/VITALS/) hosted by NASA [Land Processes Distributed Activate Archive Center (LP DAAC)](https://lpdaac.usgs.gov/) and NASA [Jet Propulsion Laboratory (JPL)](https://www.jpl.nasa.gov/).

> Hands-on exercises teaching learners how to manipluate data collected from the [ECOSTRESS](https://ecostress.jpl.nasa.gov/) and [EMIT](https://earth.jpl.nasa.gov/emit/) instruments onboard the International Space Station were delivered using the 2i2c OpenScapes Hub. [Jupyter notebooks](https://nasa.github.io/VITALS/python/01_Finding_Concurrent_Data.html) were used to demonstrate how open source tools together with cloud data and compute resources could effectively analyse the the Canopy Water Content and the Land Surface Temperature over the [Jack and Laura Dangermond Preserve](https://www.dangermondpreserve.org/), Santa Barbara, CA.

Reviewer 2: ```suggestion
Instructors used the 2i2c OpenScapes Hub to lead hands-on exercises teaching learners how to manipluate data collected from the [ECOSTRESS](https://ecostress.jpl.nasa.gov/) and [EMIT](https://earth.jpl.nasa.gov/emit/) instruments onboard the International Space Station. They used [Jupyter notebooks](https://nasa.github.io/VITALS/python/01_Finding_Concurrent_Data.html) in the Hub to demonstrate how open source tools together with cloud data and compute resources could effectively analyse the the Canopy Water Content and the Land Surface Temperature over the [Jack and Laura Dangermond Preserve](https://www.dangermondpreserve.org/), Santa Barbara, CA.
```

Minor suggestion for active voice

Author: ```suggestion
Instructors used the 2i2c Openscapes Hub to lead hands-on exercises teaching learners how to manipluate data collected from the [ECOSTRESS](https://ecostress.jpl.nasa.gov/) and [EMIT](https://earth.jpl.nasa.gov/emit/) instruments onboard the International Space Station. They used [Jupyter notebooks](https://nasa.github.io/VITALS/python/01_Finding_Concurrent_Data.html) in the Hub to demonstrate how open source tools together with cloud data and compute resources could effectively analyse the the Canopy Water Content and the Land Surface Temperature over the [Jack and Laura Dangermond Preserve](https://www.dangermondpreserve.org/), Santa Barbara, CA.
```

> Hands-on exercises teaching learners how to manipluate data collected from the [ECOSTRESS](https://ecostress.jpl.nasa.gov/) and [EMIT](https://earth.jpl.nasa.gov/emit/) instruments onboard the International Space Station were delivered using the 2i2c OpenScapes Hub. [Jupyter notebooks](https://nasa.github.io/VITALS/python/01_Finding_Concurrent_Data.html) were used to demonstrate how open source tools together with cloud data and compute resources could effectively analyse the the Canopy Water Content and the Land Surface Temperature over the [Jack and Laura Dangermond Preserve](https://www.dangermondpreserve.org/), Santa Barbara, CA.

Reviewer: ```suggestion
Hands-on exercises teaching learners how to manipluate data collected from the [ECOSTRESS](https://ecostress.jpl.nasa.gov/) and [EMIT](https://earth.jpl.nasa.gov/emit/) instruments onboard the International Space Station were delivered using the 2i2c Openscapes Hub. [Jupyter notebooks](https://nasa.github.io/VITALS/python/01_Finding_Concurrent_Data.html) were used to demonstrate how open source tools together with cloud data and compute resources could effectively analyse the the Canopy Water Content and the Land Surface Temperature over the [Jack and Laura Dangermond Preserve](https://www.dangermondpreserve.org/), Santa Barbara, CA.
```

Reviewer: Only the "O" in Openscapes is capitalized; maybe do a spell check to see if there are others I missed :)

Author: Done!

![Plot of the Canopy Water Content over the Jack and Laura Dangermond Preserve, Santa Barbara, CA.](canopy-water-content.png "Plot of the Canopy Water Content over the Jack and Laura Dangermond Preserve, Santa Barbara, CA from a [VITALS Workshop Jupyter notebook](https://nasa.github.io/VITALS/python/03_EMIT_CWC_from_Reflectance.html).")

> This event was attended by around 250 participants. To facilitate a smooth hub log-in process that didn't rely on GitHub authentication, we set up a shared password that workshop organizers can hand out to learners for access. This bypassed the manual labour of sending out GitHub invitations and drastically reduces the amount of preparatory work for events at this scale, so that organizers could focus on the essential complexity of teaching data analysis rather than the accidental complexity of managing hub authorization.

Reviewer 2: Love this highlighting of essential vs accidental complexity!

Author: ❤️

> This event was attended by around 250 participants. To facilitate a smooth hub log-in process that didn't rely on GitHub authentication, we set up a shared password that workshop organizers can hand out to learners for access. This bypassed the manual labour of sending out GitHub invitations and drastically reduces the amount of preparatory work for events at this scale, so that organizers could focus on the essential complexity of teaching data analysis rather than the accidental complexity of managing hub authorization.

Reviewer 2: If (part of) the intention of the post is to highlight the new shared-password workshop, I wonder if the lede is a bit buried? Do you want to add something to the title or opening paragraph to highlight it?

Reviewer: +1 about adding it to the title and opening paragraph

Author: Resolved in https://github.com/2i2c-org/2i2c-org.github.io/pull/274/commits/2ac81f2aee48b29c9a8cd431663521739f9ee6a5

> This event was attended by around 250 participants. To facilitate a smooth hub log-in process that didn't rely on GitHub authentication, we set up a shared password that workshop organizers can hand out to learners for access. This bypassed the manual labour of sending out GitHub invitations and drastically reduces the amount of preparatory work for events at this scale, so that organizers could focus on the essential complexity of teaching data analysis rather than the accidental complexity of managing hub authorization.

Reviewer 3: instead of 'smooth' login in process... maybe something more like instantaneous.   Even though instantaneous login was probably the experience participants expected (because its 2024), prior to the shared login in option; In advance of workshops:

1. workshop leaders retrieve participant GitHub handle (with potential added work of participant making an account) 
2. workshop leaders add participant added to Openscapes space 
3. which triggers participant receiving a sketch email from GitHub with link that expires in 7 days to be added to Openscapes Team Space
4. workshop leaders check to see how many participants need to be added to team space
5. workshop leaders still have to take time out of "hands on learning" to get everyone started in the same computation space

Reviewer 3: I think there is also a lot of _value_ in not adding to high cognitive load by having participant frustration..... that is just really fantastic :)

Reviewer: I think this is great to contrast. Could frame this as "for some longer workshops with smaller numbers of participants where a workflow with GitHub is part of the lesson, the following approach can work well. However, we were interested in a more instantaneous process for short workshops with large numbers of participants where GitHub is not part of the lesson."

Author: I've taken your comments and expanded on this point in https://github.com/2i2c-org/2i2c-org.github.io/pull/274/commits/1db469b8c8a4a3343c0634babcbf3b6e758d5696 😄

One of the elements that enabled us to recognize and solve this issue effectively is our close partnership with the Openscapes team. We engage in regular [6-weekly catch-ups](https://github.com/NASA-Openscapes/2i2cAccessPolicies/issues/7) where we can learn about user requirements and how we can develop our infrastructure to co-create optimal solutions. Together with our [Product Delivery Flow](https://team-compass.2i2c.org/product/deliveryflow/#defining-our-product-delivery-flow), we were quickly able to architect the shared password solution in time for the workshop.

![Slack message from Bri Lind](slack.png "Feedback from Bri Lind (LP DAAC)")

We have documented the technical infrastructure changes required to enable a shared password for the hub in our [Infrastructure Guide](https://infrastructure.2i2c.org/hub-deployment-guide/configure-auth/shared-password/) and hope to support many future events with this mechanism.

## Acknowledgements

> - [NASA OpenScapes](https://nasa-openscapes.github.io/)

Reviewer: ```suggestion
- [NASA Openscapes](https://nasa-openscapes.github.io/)
```
